src/useCases/getCompanies.ts

```diff
@@ -0,0 +1,36 @@
+import { prisma } from '@/lib/prisma'
+import type { Company } from '@prisma/client'
+
+interface GetCompaniesResult {
+  status: number
+  success: boolean
+  data?: Company[]
+  message?: string
+}
+
+export const getCompanies = async (): Promise<GetCompaniesResult> => {
+  try {
```

Comment:
We don't need `try/catch` block because all errors are handled by API endpoint :-)